bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/core/provider/ModuleTypeResourceBundleProvider.java

```diff
@@ -0,0 +1,135 @@
+ *
+ */
+public class ModuleTypeResourceBundleProvider extends GeneralResourceBundleProvider implements ModuleTypeProvider {
+
+    /**
+     * @param bc
+     */
+    public ModuleTypeResourceBundleProvider(BundleContext bc) {
+        super(bc);
+        path = PATH + "/module.types/";
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.handler.provider.StreamModuleTypeProvider#remove(java.net.URI)
+     */
+    public void remove(URL url) {
+        synchronized (lock) {
+            List portfolio = providerPortfolio.get(url.toString());
+            if (portfolio == null || portfolio.isEmpty())
+                return;
+            Iterator i = portfolio.iterator();
+            while (i.hasNext()) {
+                String uid = (String) i.next();
+                providedObjectsHolder.remove(uid);
+            }
+        }
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.handler.ModuleTypeResourceBundleProvider#getModuleType(java.lang.String,
+     *      java.util.Locale)
+     */
+    @Override
+    public ModuleType getModuleType(String UID, Locale locale) {
+        Localizer l = null;
+        synchronized (lock) {
+            l = providedObjectsHolder.get(UID);
+        }
+        if (l != null) {
+            ModuleType mt = (ModuleType) l.localize(locale);
+            return mt;
+        }
+        return null;
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.handler.ModuleTypeResourceBundleProvider#getModuleTypes(java.util.Locale)
+     */
+    @Override
+    public Collection<ModuleType> getModuleTypes(Locale locale) {
+        ArrayList moduleTypesList = new ArrayList();
+        synchronized (lock) {
+            Iterator i = providedObjectsHolder.values().iterator();
+            while (i.hasNext()) {
+                Localizer l = (Localizer) i.next();
+                if (l != null) {
+                    ModuleType mt = (ModuleType) l.localize(locale);
+                    if (mt != null)
+                        moduleTypesList.add(mt);
+                }
+            }
+        }
+        return moduleTypesList;
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.core.provider.GeneralResourceBundleProvider#addingService(ServiceReference)
+     */
+    @Override
+    public Object addingService(ServiceReference reference) {
+        if (reference.getProperty(Parser.PARSER_TYPE).equals(Parser.PARSER_MODULE_TYPE)) {
+            return super.addingService(reference);
+        }
+        return null;
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.core.provider.GeneralResourceBundleProvider#getUID(java.lang.Object)
+     */
+    @Override
+    protected String getUID(Object providedObject) {
+        return ((ModuleType) providedObject).getUID();
+    }
+
+    protected Object getKey(Object element) {
+        ModuleType mt = (ModuleType) element;
+        return mt.getUID();
+    }
+
+    protected String getStorageName() {
```

Comment:
What are getStorageName, keyToString, toElement and toPersistableElement good for? They are not used and I do not see their purpose.